numpy/lib/_index_tricks_impl.py

```diff
@@ -683,15 +683,14 @@ class ndindex:
     (2, 0, 0)
     (2, 1, 0)
 
-    """
-
+     """
     def __init__(self, *shape):
         if len(shape) == 1 and isinstance(shape[0], tuple):
             shape = shape[0]
-        x = as_strided(_nx.zeros(1), shape=shape,
-                       strides=_nx.zeros_like(shape))
-        self._it = _nx.nditer(x, flags=['multi_index', 'zerosize_ok'],
-                              order='C')
+        if any(dim < 0 for dim in shape):
```

Comment:
```suggestion
        if min(shape) < 0:
```
Is perhaps faster, as it avoids creation of a generator.